api/src/test/java/org/pmiops/workbench/api/ClusterControllerTest.java

```diff
@@ -122,6 +126,9 @@ public void setUp() {
     when(userProvider.get()).thenReturn(user);
     clusterController.setUserProvider(userProvider);
 
+    UserService userService = new UserService(userProvider, userDao, adminActionHistoryDao, clock, fireCloudService, configProvider);
```

Comment:
Yeah, this is probably the best you can do here for now. The ugliness stems from me not being able to figure out how to inject a provider with a stubbable user (it can probably be done). So the controller would initially be created using the mock `UserService` defined via `@MockBean` above, then you replace it here.

Reply:
Yeah, exactly. That's the thought at least.